lib/elements/aws.go

```diff
@@ -0,0 +1,88 @@
+package elements
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+)
+
+func getData(url string) (data []string) {
+	resp, err := http.Get(url)
+	if err != nil {
+		return make([]string, 0)
+	}
+
+	defer resp.Body.Close()
+
+	scanner := bufio.NewScanner(resp.Body)
+	for scanner.Scan() {
+		data = append(data, strings.TrimRight(scanner.Text(), "\n"))
+		if err != nil {
+			break
+		}
+	}
+	return
+}
+
+func crawlData(url string) map[string]interface{} {
+	data := make(map[string]interface{})
+	urlData := getData(url)
+
+	var key string
+	for _, line := range urlData {
+
+		// replace hyphens with underscores for JSON keys
+		key = strings.Replace(line, "-", "_", -1)
+
+		switch {
+		default:
+			data[key] = getData(url + line)[0]
```

Comment:
Actually, spotted something here. I did some quick modifications for OpenStack, and it crashes at this point. If I change the above to the following, everything works:

```go
    default:                   
      d := getData(url + line) 
      if len(d) > 0 {          
        data[key] = d[0]       
      }                        
```